pkg/stash/with_etcd.go

```diff
@@ -56,33 +56,31 @@ func (s *etcd) Stash(ctx context.Context, mod, ver string) (newVer string, err e
 	const op errors.Op = "etcd.Stash"
 	ctx, span := observ.StartSpan(ctx, op.String())
 	defer span.End()
-	sesh, err := concurrency.NewSession(s.client)
+
+	sess, err := concurrency.NewSession(s.client)
 	if err != nil {
-		return ver, errors.E(op, err)
+		return "", errors.E(op, err)
 	}
-	mv := config.FmtModVer(mod, ver)
-	mu := concurrency.NewMutex(sesh, mv)
-	err = mu.Lock(ctx)
-	if err != nil {
-		return ver, errors.E(op, err)
+	defer sess.Close()
+
+	m := concurrency.NewMutex(sess, config.FmtModVer(mod, ver))
+	if err := m.Lock(ctx); err != nil {
+		return "", errors.E(op, err)
 	}
-	defer func() {
```

Comment:
I prefer keeping it the way it was.
https://go.dev/play/p/cq0DGy9OJES

Knowing if unlock / session close fails is a useful thing

Reply:
The reason I made this change is because it seems a bit strange to fail the request because the mutex failed to unlock. There's not really any action to be taken, and everything else has already succeeded. The best thing we can do here is log, not fail the entire request - as it doesn't matter.

Reply:
I agree; I think we should definitely log. Can we have some tests if possible please

Reply:
@manugupt1 I've replaced the etcd tests. They should now be a lot more realistic and actually connects to an etcd server. The tests cover the main code paths and ensure that a lease is acquired, and then freed again.

Reply:
Sorry for asking again, can we log if unlock or session close fails?

Reply:
@manugupt1 There doesn't really seem to be a standard for logging in this repository. What should I use?

Reply:
I also am not entirely sure it's worth logging. It _could_ be useful? Though neither of these conditions should have any long-lasting negative effects. If the connection to etcd is unhealthy then the lease will expire automatically with no hard done.

Reply:
@manugupt1 What are your thoughts? I'm also keen to merge this one as it is quite a major issue for the etcd integration.

I am sort of surprised this issue wasn't raised before - I wonder how many people use Athens with etcd.